ddl/ddl_api.go

```diff
@@ -932,6 +956,36 @@ func (d *ddl) RebaseAutoID(ctx context.Context, ident ast.Ident, newBase int64)
 	return errors.Trace(err)
 }
 
+func (d *ddl) ShardRowID(ctx context.Context, ident ast.Ident, uVal uint64) error {
+	job, err := d.createJobByIdent(ident)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	job.Type = model.ActionShardRowID
+	job.Args = []interface{}{uVal}
+	err = d.doDDLJob(ctx, job)
+	err = d.callHookOnChanged(err)
+	return errors.Trace(err)
+}
+
+func (d *ddl) createJobByIdent(ident ast.Ident) (*model.Job, error) {
```

Comment:
createJobByName is a better name?

Reply:
`ident` has both `Schema` and `TableName`, it's not a single name.

Reply:
createJobForTable

Reply:
createJobForTable is a better name.